### PR TITLE
Add basic bzlmod example

### PR DESCRIPTION
**Pull request by keith**

### CODE REVIEW

**Comment by keith** [comment]
The first commit is https://github.com/bazelbuild/rules_android/pull/147, second one duplicates basicapp and makes it bzlmod compatible

File: examples/basicappbzlmod/.bazelrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+common --enable_bzlmod --lockfile_mode=off
+
+# Disabled to avoid //external:android deps in bazel itself
+common --android_sdk=@androidsdk//:sdk
+common --optimizing_dexer=//:optimizing_dexer
```
Comment (keith) [inline examples/basicappbzlmod/.bazelrc:5]: this is a workaround to avoid having this target in bazel reference `//external:android`

**Comment by keith** [comment]
Using this bazel https://github.com/bazelbuild/bazel/pull/19575 you can almost get a working build of the sample app added here, in general this gives an idea of what needs to be fixed

**Comment by ted-xie** [comment]
We ended up independently implementing this in the regular basic example app a while ago.

**Pull request closed without merging**